### PR TITLE
Hw 02 git

### PR DESCRIPTION
**Pull request by SichniyAndriy**

### CODE REVIEW

File: src/main/java/core/basesyntax/Application.java [hunk 1]
```diff
@@ -3,5 +3,13 @@
...
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
+        Lottery lot = new Lottery();
+        Ball[] balls = new Ball[3];
+        for (int i = 0; i < balls.length; ++i) {
+            balls[i] = lot.getRandomBall();
+        }
+        for (Ball ball: balls) {
+            System.out.println(ball.toString());
+        }
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Application.java:13]: lets use just 1 foor-loop

File: src/main/java/core/basesyntax/Ball.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        return number;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    @Override
+    public String toString() {
+        return "Number: " + this.getNumber() + ", Color: " + this.getColor();
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Ball.java:22]:
```suggestion
        return "Number: " + number + ", Color: " + color;
```
you have direct access

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 3]
```diff
@@ -1,7 +1,10 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
     public String getRandomColor() {
-        return null;
+        int i = new Random().nextInt(ColorsCollection.values().length);
```
Comment (aaananas) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: lets make Random global, create a field in class

File: src/main/java/core/basesyntax/ColorsCollection.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+public enum ColorsCollection {
```
Comment (aaananas) [inline src/main/java/core/basesyntax/ColorsCollection.java:3]:
```suggestion
public enum Color {

```

File: src/main/java/core/basesyntax/ColorsCollection.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package core.basesyntax;
+
+public enum ColorsCollection {
+    Red,
+    Yellow,
+    Green,
+    White,
+    Black,
+    Blue,
+    Purple
```
Comment (aaananas) [inline src/main/java/core/basesyntax/ColorsCollection.java:10]: enum values should be named in uppercase

File: src/main/java/core/basesyntax/Lottery.java [hunk 6]
```diff
@@ -0,0 +1,9 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    public Ball getRandomBall() {
+        return new Ball(new Random().nextInt(100), new ColorSupplier().getRandomColor());
```
Comment (aaananas) [inline src/main/java/core/basesyntax/Lottery.java:7]: lets make random and colorSupplier global

File: src/main/java/core/basesyntax/Application.java (hunk 1 above)
Comment (SichniyAndriy) [reply src/main/java/core/basesyntax/Application.java:13]: ok

File: src/main/java/core/basesyntax/Application.java [hunk 7]
```diff
@@ -3,5 +3,11 @@
 public class Application {
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
+        Lottery lot = new Lottery();
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/Application.java:6]:
```suggestion
        Lottery lottery = new Lottery();
```

File: src/main/java/core/basesyntax/Application.java [hunk 8]
```diff
@@ -3,5 +3,11 @@
...
+        Ball[] balls = new Ball[3];
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/Application.java:7]: Why add these values to an array when you can just print them?

File: src/main/java/core/basesyntax/Application.java [hunk 9]
```diff
@@ -3,5 +3,11 @@
...
+        for (int i = 0; i < balls.length; ++i) {
+            balls[i] = lot.getRandomBall();
+            System.out.println(balls[i].toString());
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/Application.java:10]:
```suggestion
            System.out.println(lot.getRandomBall());
```

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 10]
```diff
@@ -1,7 +1,12 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
+    private Random rand;
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]:
```suggestion
    private Random random;
```
Why abbreviate names? It makes it harder to read

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 11]
```diff
@@ -1,7 +1,12 @@
...
+
     public String getRandomColor() {
-        return null;
+        int i = rand.nextInt(Color.values().length);
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/ColorSupplier.java:9]:
```suggestion
        int i = rand.nextInt(Color.values().length);
```
i is not informative name

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 12]
```diff
@@ -1,7 +1,12 @@
...
+        return Color.values()[i].toString();
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/ColorSupplier.java:10]:
```suggestion
        return Color.values()[i].name();
```
read checklist more carefully
Screen Shot 2022-10-17 at 8 35 17 PM

File: src/main/java/core/basesyntax/Lottery.java [hunk 13]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    private Random rand;
+    private ColorSupplier colorSupplier;
+
+    public Ball getRandomBall() {
+        return new Ball(rand.nextInt(100), colorSupplier.getRandomColor());
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/Lottery.java:10]: 100 must be a constant. Also create variables for rand.nextInt(100) and colorSupplier.getRandomColor(). This makes the code more readable

File: src/main/java/core/basesyntax/Lottery.java [hunk 14]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    private Random rand;
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/Lottery.java:6]:
```suggestion
    private Random random;

```

**Review by zachar-chorny — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Application.java [hunk 15]
```diff
@@ -3,5 +3,10 @@
 public class Application {
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
+        final int amount = 3;
+        Lottery lottery = new Lottery();
+        for (int i = 0; i < amount; ++i) {
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/Application.java:8]: amount is not a constant. See what constants are

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 16]
```diff
@@ -1,7 +1,16 @@
...
 public class ColorSupplier {
+    private final Random random;
+
+    public ColorSupplier() {
+        random = new Random();
+    }
+
     public String getRandomColor() {
-        return null;
+        int i = random.nextInt(Color.values().length);
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/ColorSupplier.java:13]: Didn't correct my comment about the variable name

File: src/main/java/core/basesyntax/Lottery.java [hunk 17]
```diff
@@ -0,0 +1,18 @@
...
+    private final Random random;
+    private final ColorSupplier colorSupplier;
+
+    public Lottery() {
+        random = new Random();
+        colorSupplier = new ColorSupplier();
+    }
+
+    public Ball getRandomBall() {
+        final int max = 100;
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/Lottery.java:15]: max is not a constant

File: src/main/java/core/basesyntax/Lottery.java [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+        return new Ball(random.nextInt(max), colorSupplier.getRandomColor());
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/Lottery.java:16]: This comment is ignored 'Also create variables for rand.nextInt(100) and colorSupplier.getRandomColor(). This makes the code more readable'

**Review by zachar-chorny — CHANGES_REQUESTED** [review]
Don't ignore the comments! Also, why are commits from another account going into your branch? 
Screen Shot 2022-10-17 at 9 56 29 PM

**Review by zachar-chorny — CHANGES_REQUESTED** [review]
I have already written the comments above

**Review by zachar-chorny — CHANGES_REQUESTED** [review]
What are you doing? Firstly fix comments above

**Comment by SichniyAndriy** [comment]
Взагалі не розумію звідки він взявся

**Comment by SichniyAndriy** [comment]
image
він у мене в учасниках

**Comment by SichniyAndriy** [comment]
image
всі комміти та пуші роблю Bash на моєму компі

**Comment by SichniyAndriy** [comment]
а вилазить якась інша аватарка

**Pull request closed without merging**